.asf.yaml

```diff
@@ -0,0 +1,42 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# https://cwiki.apache.org/confluence/display/INFRA/Git+-+.asf.yaml+features
+
+notifications:
+  commits: commits@arrow.apache.org
+  issues: issues@arrow.apache.org
+  pullrequests: github@arrow.apache.org
+
+github:
+  description: "Apache Arrow Julia"
+  homepage: https://arrow.apache.org/
+  labels:
+    - apache-arrow
+    - julia
+  features:
+    issues: true
+  enabled_merge_buttons:
+    squash: true
+    merge: false
+    rebase: false
+  protected_branches:
+    main:
+      required_linear_history: true
+      required_pull_request_reviews:
+        dismiss_stale_reviews: true
+        required_approving_review_count: 1
```

Comment:
@quinnj It seems that JuliaData/Arrow.jl uses protected branch feature but I don't know the configuration. Are there some missing configurations?

See also https://docs.github.com/en/rest/reference/branches#update-branch-protection for available configurations.

Reply:
We also had "Automatically delete head branches " feature enabled, but I think that's the only one I don't see in the list here.

Reply:
Thanks.
It seems that we can enable the feature by https://docs.github.com/en/rest/reference/repos#update-a-repository API.
But the `.asf.yaml` mechanism doesn't support it yet: https://github.com/apache/infrastructure-p6/blob/production/modules/gitbox/files/asfgit/package/asfyaml.py#L234
It seems that the `.asf.yaml` mechanism doesn't change the current configuration. I hope that we can still use the feature. We can confirm it by merging this pull request.